src/btree/bt_sync.c

```diff
@@ -383,7 +387,8 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 		__wt_txn_release_snapshot(session);
 
 	/* Clear the checkpoint flag. */
-	btree->checkpointing = WT_CKPT_OFF;
+	btree->syncing = WT_BTREE_SYNC_OFF;
+	btree->sync_session = NULL;
```

Comment:
Is the order important here? Do we need to make sure that syncing state change is visible to other threads before clearing `sync_session`?

Reply:
I don't think it matters -- the value of `sync_session` is only relevant if it is equal to `session`: in other words, it's only the current thread that will do anything different depending on the value, so the order of operations here isn't important.

It's also worth noting that this code already assumes that there is some higher level concurrency control preventing multiple threads from syncing the same tree at the same time.  So if you're worried that some other thread will flip `syncing` to `WT_BTREE_SYNC_RUNNING` and have its `sync_session` cleared, I don't think that's possible.

All of that said, if the code reads better with manipulation of `sync_session` nested inside `syncing` (or vice versa), I'm fine with that.

Reply:
Thanks for the explain - makes sense, and I don't think it requires further changes for clarity. I'll go ahead and merge.